tests/test_core.py

```diff
@@ -2419,6 +2419,16 @@ def test_pthread_specific(self):
   def test_pthread_equal(self):
     self.do_run_in_out_file_test('pthread/test_pthread_equal.cpp')
 
+  @node_pthreads
+  def test_pthread_proxying(self):
+    self.set_setting('EXIT_RUNTIME')
+    self.set_setting('PROXY_TO_PTHREAD')
+    self.set_setting('PTHREAD_POOL_SIZE=3')
```

Comment:
I think only one of PROXY_TO_PTHREAD and PTHREAD_POOL_SIZE should be needed (i.e. you don't need a pool if you are proxying your main).